actions_test.go

```diff
@@ -0,0 +1,160 @@
+package opslevel_test
+
+import (
+	ol "github.com/opslevel/opslevel-go/v2023"
+	"github.com/rocktavious/autopilot/v2022"
+	"testing"
+)
+
+func TestWebhookActionHeadersMarshalling(t *testing.T) {
+	// Arrange
+	unstructured := "Accept=\"application/vnd.github+json\"\nAuthorization=\"Bearer XXX\"\nContent-Type=\"application/json\"\nHost=\"app.opslevel.com\"\nOrigin=\"api.opslevel.com\""
+	structured := map[string]string{
+		"Host":          "app.opslevel.com",
+		"Origin":        "api.opslevel.com",
+		"Content-Type":  "application/json",
+		"Accept":        "application/vnd.github+json",
+		"Authorization": "Bearer XXX",
+	}
+	// Act
+	marshalled := ol.CustomActionsToHeaders(unstructured)
+	unmarshalled := ol.CustomActionsFromHeaders(structured)
+	// Assert
+	autopilot.Equals(t, structured, marshalled)
+	autopilot.Equals(t, unstructured, unmarshalled)
+}
+
+func TestCreateWebhookAction(t *testing.T) {
+	//Arrange
+	request := `{"query": 
+		"mutation($input:CustomActionsWebhookActionCreateInput!){customActionsWebhookActionCreate(input: $input){webhookAction{{ template "custom_actions_request" }},errors{message,path}}}",
+		"variables":{"input":{"headers":"Content-Type=\"application/json\"","httpMethod":"POST","liquidTemplate":"{\"token\": \"XXX\", \"ref\":\"main\", \"action\": \"rollback\"}","name":"Deploy Rollback","webhookUrl":"https://gitlab.com/api/v4/projects/1/trigger/pipeline"}}
+	}`
+	response := `{"data": {"customActionsWebhookActionCreate": {
+        "webhookAction": {{ template "custom_action1" }},
+        "errors": []
+    }}}`
+
+	client := ABetterTestClient(t, "custom_actions/create_action", request, response)
+	// Act
+	trigger, err := client.CreateWebhookAction(ol.CustomActionsWebhookActionCreateInput{
+		Name:           "Deploy Rollback",
+		LiquidTemplate: "{\"token\": \"XXX\", \"ref\":\"main\", \"action\": \"rollback\"}",
+		Headers: map[string]string{
+			"Content-Type": "application/json",
+		},
+		HTTPMethod: ol.CustomActionsHttpMethodEnumPost,
+		WebhookURL: "https://gitlab.com/api/v4/projects/1/trigger/pipeline",
+	})
+	// Assert
+	autopilot.Ok(t, err)
+	autopilot.Equals(t, "Deploy Rollback", trigger.Name)
+}
+
+func TestListCustomActions(t *testing.T) {
+	//Arrange
+	request := `{"query":
+		"{account{customActionsExternalActions(after: $after, first: $first){nodes{{ template "custom_actions_request" }},{{ template "pagination_request" }}}}}",
+		"variables":{}
+	}`
+	response := `{"data": {"account": {
+        "customActionsExternalActions": {
+          "nodes": [
+            {{ template "custom_action1" }},
+            {{ template "custom_action2" }}
+          ],
+          {{ template "no_pagination_response" }},
+          "totalCount": 2
+        }
+    }}}`
+
+	// An easy way to see the results of templating is by uncommenting this
+	//fmt.Print(Templated(request))
+	//fmt.Print(Templated(response))
+	//panic(1)
```

Comment:
I left this in here as an example incase you need to debug the templated data that gets generated for the request and response